### ISSUE TITLE
10 tests fail

### ISSUE THREAD

**Issue by yurivict** [jsonpickle/jsonpickle#492]
```
========================================================================================= FAILURES ==========================================================================================
___________________________________________________________________________________ jsonpickle/version.py ___________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
9: error: Incompatible types in assignment (expression has type "None", variable has type Module)  [assignment]
_________________________________________________________________________________ jsonpickle/ext/pandas.py __________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
4: error: Library stubs not installed for "pandas"  [import-untyped]
4: note: Hint: "python3 -m pip install pandas-stubs"
4: note: (or run "mypy --install-types" to install all missing stub packages)
_______________________________________________________________________________________ test session ________________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
mypy exited with status 1.
__________________________________________________________________________________ jsonpickle/ext/gmpy.py ___________________________________________________________________________________
[gw0] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
2: error: Skipping analyzing "gmpy2": module is installed, but missing library stubs or py.typed marker  [import-untyped]
2: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
_______________________________________________________________________________________ docs/conf.py ________________________________________________________________________________________
[gw3] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
5: error: Skipping analyzing "furo": module is installed, but missing library stubs or py.typed marker  [import-untyped]
9: error: Cannot find implementation or library stub for module named "rst.linker"  [import-not-found]
9: error: Cannot find implementation or library stub for module named "rst"  [import-not-found]
____________________________________________________________________________________ tests/ecdsa_test.py ____________________________________________________________________________________
[gw1] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
27: error: Skipping analyzing "ecdsa": module is installed, but missing library stubs or py.typed marker  [import-untyped]
28: error: Skipping analyzing "ecdsa.keys": module is installed, but missing library stubs or py.typed marker  [import-untyped]
___________________________________________________________________________________ tests/pandas_test.py ____________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
7: error: Library stubs not installed for "pandas"  [import-untyped]
8: error: Library stubs not installed for "pandas.testing"  [import-untyped]
_________________________________________________________________________________ tests/feedparser_test.py __________________________________________________________________________________
[gw1] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
77: error: Skipping analyzing "feedparser": module is installed, but missing library stubs or py.typed marker  [import-untyped]
___________________________________________________________________________________ tests/sklearn_test.py ___________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
5: error: Skipping analyzing "sklearn.datasets": module is installed, but missing library stubs or py.typed marker  [import-untyped]
5: error: Skipping analyzing "sklearn": module is installed, but missing library stubs or py.typed marker  [import-untyped]
6: error: Skipping analyzing "sklearn.linear_model": module is installed, but missing library stubs or py.typed marker  [import-untyped]
7: error: Skipping analyzing "sklearn.tree": module is installed, but missing library stubs or py.typed marker  [import-untyped]
_________________________________________________________________________________ tests/sqlalchemy_test.py __________________________________________________________________________________
[gw2] freebsd14 -- Python 3.9.18 /usr/local/bin/python3.9
10: error: Skipping analyzing "sqlalchemy": module is installed, but missing library stubs or py.typed marker  [import-untyped]
12: error: Skipping analyzing "sqlalchemy.ext": module is installed, but missing library stubs or py.typed marker  [import-untyped]
13: error: Skipping analyzing "sqlalchemy.orm": module is installed, but missing library stubs or py.typed marker  [import-untyped]
27: error: Variable "sqlalchemy_test.Base" is not valid as a type  [valid-type]
27: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
27: error: Invalid base class "Base"  [misc]
===================================================================================== warnings summary ======================================================================================
../../../../../local/lib/python3.9/site-packages/xdist/plugin.py:252
../../../../../local/lib/python3.9/site-packages/xdist/plugin.py:252
../../../../../local/lib/python3.9/site-packages/xdist/plugin.py:252
../../../../../local/lib/python3.9/site-packages/xdist/plugin.py:252
../../../../../local/lib/python3.9/site-packages/xdist/plugin.py:252
  /usr/local/lib/python3.9/site-packages/xdist/plugin.py:252: DeprecationWarning: The --rsyncdir command line argument and rsyncdirs config variable are deprecated.
  The rsync feature will be removed in pytest-xdist 4.0.
    config.issue_config_time_warning(warning, 2)

../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:147: 111 warnings
  /usr/local/lib/python3.9/site-packages/_pytest/nodes.py:147: PytestDeprecationWarning: <class 'pytest_black.BlackItem'> is not using a cooperative constructor and only takes {'parent', 'fspath'}.
  See https://docs.pytest.org/en/stable/deprecations.html#constructors-of-custom-pytest-node-subclasses-should-take-kwargs for more details.
    warnings.warn(

../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:264: 148 warnings
  /usr/local/lib/python3.9/site-packages/_pytest/nodes.py:264: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to BlackItem is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return cls._create(parent=parent, **kw)

../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:724
../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:724
../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:724
../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:724
  /usr/local/lib/python3.9/site-packages/_pytest/nodes.py:724: PytestWarning: BlackItem is an Item subclass and should not be a collector, however its bases File are collectors.
  Please split the Collectors and the Item into separate node types.
  Pytest Doc example: https://docs.pytest.org/en/latest/example/nonpython.html
  example pull request on a plugin: https://github.com/asmeurer/pytest-flakes/pull/40/
    warnings.warn(

../../../../../local/lib/python3.9/site-packages/_pytest/nodes.py:147: 37 warnings
  /usr/local/lib/python3.9/site-packages/_pytest/nodes.py:147: PytestDeprecationWarning: <class 'pytest_black.BlackItem'> is not using a cooperative constructor and only takes {'fspath', 'parent'}.
  See https://docs.pytest.org/en/stable/deprecations.html#constructors-of-custom-pytest-node-subclasses-should-take-kwargs for more details.
    warnings.warn(

tests/numpy_test.py::test_ndarray_roundtrip
tests/sklearn_test.py::test_decision_tree
  /usr/ports/devel/py-jsonpickle/work-py39/jsonpickle-3.0.4/jsonpickle/ext/numpy.py:305: UserWarning: ndarray is defined by reference to an object we do not know how to serialize. A deep copy is serialized instead, breaking memory aliasing.
    warnings.warn(msg)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

-------- coverage: platform freebsd14, python 3.9.18-final-0 ---------
Name                         Stmts   Miss  Cover   Missing
----------------------------------------------------------
jsonpickle/__init__.py          20      0   100%
jsonpickle/backend.py          108     19    82%   19, 36-37, 44-46, 62-63, 68-72, 127, 161-162, 168-169, 243-244
jsonpickle/compat.py            11      0   100%
jsonpickle/errors.py             3      0   100%
jsonpickle/ext/__init__.py       0      0   100%
jsonpickle/ext/gmpy.py          18     18     0%   1-27
jsonpickle/ext/numpy.py        183     10    95%   28, 70-73, 76, 107, 174, 306-311
jsonpickle/ext/pandas.py       163      0   100%
jsonpickle/handlers.py         135      6    96%   111, 118, 156, 178, 245, 288
jsonpickle/pickler.py          394     41    90%   270-274, 410-411, 416, 419, 430, 439, 458, 553, 575-579, 587-591, 603-605, 663-664, 673, 685-686, 691, 705-706, 742, 751, 758-767, 822, 855-856, 866
jsonpickle/tags.py              22      0   100%
jsonpickle/unpickler.py        456     48    89%   84, 156-157, 160, 235, 250, 252-253, 269-271, 287, 306, 455, 485, 502-513, 535-536, 541, 615, 643-658, 752, 759-764, 838, 845, 879, 896-897
jsonpickle/util.py             179     12    93%   99, 204, 209, 332, 348-349, 363-364, 402, 406-409, 443
jsonpickle/version.py           15      5    67%   5, 8-9, 16-17
----------------------------------------------------------
TOTAL                         1707    159    91%

=========================================================================================== mypy ============================================================================================
Found 20 errors in 9 files (checked 37 source files)
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/ecdsa_test.py:36: gmpy was not detected, please try installing it for more complete tests!
====================================================== 10 failed, 385 passed, 1 skipped, 3 xpassed, 307 warnings in 461.75s (0:07:41) =======================================================
*** Error code 1
```

Version: 3.0.4
Python-3.9
FreeBSD 14.0

**Comment by Theelx** [comment]
~~Did you try running ``pip install -r requirements-dev.txt``? Those are necessary for the tests that involve third-party modules.~~
@davvid I just noticed that requirements-dev.txt is no longer one of the top-level files in the main branch. Did it accidentally get removed with the garden code?

**Comment by Theelx** [comment]
Wait never mind, I just saw the commit that moved it into setup.cfg. Can we add instructions for people to install the dev requirements without needing to download ``vx``? I'll look at that now.

**Comment by stefan6419846** [comment]
`vx` should not be required, but just a regular virtual environment for isolating the packages and its dependencies. `pip install .[testing]` should be sufficient to install everything from a git checkout.

**Comment by Theelx** [comment]
Yeah, I just tried that locally and it works. I'll make add that to the README shortly. In the meantime, I wonder if there's a way to hook it so `pip install -r requirements-dev.txt` can install from setup.cfg, to avoid it breaking for normal users until an actual release like 3.1.0.

**Comment by davvid** [comment]
I just added a stub `requirements-dev.txt` for now and I also added another section in there that describes how to use a virtualenv w/out vx. Thanks for the heads-up ~ I didn't realize anyone was using `requirements-dev.txt` directly.

If the `.[dev]` line in the stub `requirements-dev.txt` isn't good enough we can always restore the original. It just seemed better to have only a single place where dependencies are specified rather than having to keep the disparate lists in sync.
